src/test/java/com/commercetools/project/sync/state/StateSyncerTest.java

```diff
@@ -1,116 +1,197 @@
-  //
-  //    // assertion
-  //    final LoggingEvent errorLog = syncerTestLogger.getAllLoggingEvents().get(1);
-  //    assertThat(errorLog.getMessage())
-  //        .isEqualTo(
-  //            "Error when trying to sync state. Existing key: <<not present>>. Update actions:
-  // []");
-  //    assertThat(errorLog.getThrowable().get().getMessage())
-  //        .isEqualTo(
-  //            "StateDraft with name: LocalizedString(en -> State 1) doesn't have a key. Please
-  // make sure all state drafts have keys.");
-  //  }
+
+  @Test
+  void of_ShouldCreateStateSyncerInstance() {
+    // test
+    final StateSyncer stateSyncer =
+        StateSyncer.of(mock(ProjectApiRoot.class), mock(ProjectApiRoot.class), getMockedClock());
+
+    // assertions
+    assertThat(stateSyncer).isNotNull();
+    assertThat(stateSyncer.getSync()).isInstanceOf(StateSync.class);
+  }
+
+  @Test
+  void transform_ShouldReplaceStateTransitionIdsWithKeys() {
+    // preparation
+    final ProjectApiRoot sourceClient = mock(ProjectApiRoot.class);
+    mockResourceIdsGraphQlRequest(
+        sourceClient, "states", "ab949f1b-c441-4c70-9cf0-4182c36d6a6c", "Initial");
+
+    final StateSyncer stateSyncer =
+        StateSyncer.of(sourceClient, mock(ProjectApiRoot.class), getMockedClock());
+    final List<State> states =
+        List.of(
+            readObjectFromResource("state-1.json", State.class),
+            readObjectFromResource("state-2.json", State.class));
+
+    // test
+    final CompletionStage<List<StateDraft>> stateDrafts = stateSyncer.transform(states);
+
+    // assertions
+    final StateDraft draft1 =
+        StateDraftBuilder.of()
+            .key("State 1")
+            .type(StateTypeEnum.LINE_ITEM_STATE)
+            .roles(List.of())
+            .description(LocalizedString.ofEnglish("State 1"))
+            .name(LocalizedString.ofEnglish("State 1"))
+            .initial(true)
+            .build();
+    final StateDraft draft2 =
+        StateDraftBuilder.of()
+            .key("State 2")
+            .type(StateTypeEnum.LINE_ITEM_STATE)
+            .roles(List.of())
+            .description(LocalizedString.ofEnglish("State 2"))
+            .name(LocalizedString.ofEnglish("State 2"))
+            .initial(false)
+            .transitions(StateResourceIdentifierBuilder.of().key("Initial").build())
+            .build();
+
+    assertThat(stateDrafts).isCompletedWithValue(List.of(draft1, draft2));
+  }
+
+  @Test
+  void getQuery_ShouldBuildStateQuery() {
+    // preparation
+
+    final ProjectApiRoot apiRoot =
+        ApiRootBuilder.of().withApiBaseUrl("apiBaseUrl").build("testProjectKey");
+    final StateSyncer stateSyncer =
+        StateSyncer.of(apiRoot, mock(ProjectApiRoot.class), getMockedClock());
+
+    // test
+    final ByProjectKeyStatesGet query = stateSyncer.getQuery();
+
+    // assertion
+    assertThat(query).isInstanceOf(ByProjectKeyStatesGet.class);
```

Comment:
This test is not useful. Since you already assigned type to the line above `final ByProjectKeyStatesGet query...` this is already ensured.

Reply:
[ea6e371](https://github.com/commercetools/commercetools-project-sync/pull/512/commits/ea6e3714c9c4ac6f947bd904800674d505d6f7d5)